include/misc/rb.h

```diff
@@ -139,6 +140,7 @@ static inline void rb_walk(struct rbtree *tree, rb_visit_t visit_fn,
 {
 	_rb_walk(tree->root, visit_fn, cookie);
 }
+#endif
```

Comment:
shouldn't `_rb_walk()` be inside this `ifdef` ?

Reply:
If you like.  For readability I just wrapped the public APIs, not the helper code (because finding matching #endif's gets annoying).

Reply:
the problem is that symbol will be exported and this will lead to two possible problems:
1) scan analyzers will spot it and we'll need to justfiy
2) Code that is never used is available, what in MISRA-C terms means dead code.